examples/angular/src/app/pages/services/editable-page.service.ts

```diff
@@ -0,0 +1,109 @@
+import { DestroyRef, inject, Injectable, signal } from '@angular/core';
+import { ActivatedRoute, NavigationEnd, Router } from '@angular/router';
+import { DYNAMIC_COMPONENTS } from '../components';
+import { DotCMSPageAsset, DotCMSPageRequestParams } from '@dotcms/types';
+import { filter, map, startWith, switchMap, tap } from 'rxjs/operators';
+import { PageService } from './page.service';
+import { DotCMSEditablePageService } from '@dotcms/angular/next';
+import { BASE_EXTRA_QUERIES } from '../../shared/constants';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { getUVEState } from '@dotcms/uve';
+import { ComposedPageResponse, ExtraContent, PageRender } from '../../shared/models';
+import { Observable } from 'rxjs';
+
+/**
+ * Service that handles page loading and management for DotCMS pages
+ * This service should be provided at the component level, not in root,
+ * to ensure each page component has its own isolated instance.
+ */
+@Injectable()
+export class EditablePageService<
+    TPage extends DotCMSPageAsset = DotCMSPageAsset,
+    TContent = ExtraContent
```

Comment:
Would it make sense to have this receive a composed page type?

We could then create a composed type for each kind of page. For example:

```ts
export class EditablePageService<ComposedPage> {}
```

And define the types like:

```ts
type GenericPage = ComposedPage<T, P>;
type BlogPage = ComposedPage<T, P>;
```

Which would allow usage like:

```ts
const service = EditablePageService<GenericPage>;
const service2 = EditablePageService<BlogPage>;
```

This could give us more flexibility, stronger, and cleaner typing per page type. What do you think?

Reply:
I like it, will change it